packages/langium-sql/src/sql-type-utilities.ts

```diff
@@ -229,3 +229,216 @@ function getColumnsForTableSource(source: AST.TableSource): ColumnDescriptor[] {
+            'TEXT',
+            'NCHAR',
+            'NVARCHAR',
+            'NTEXT'
+        ]
+    };
+
+    export const INTEGERS: DataTypeList = {
+        SQL: [
+            'INT',
+            'INTEGER'
+        ],
+        MySQL: [
+            'BIT(size)',
+            'TINYINT(size)',
+            'SMALLINT(size)',
+            'MEDIUMINT(size)',
+            'INT(size)',
+            'INTEGER(size)',
+            'BIGINT(size)'
+        ],
+        Oracle: [],
+        SQLServer: [
+            'BIT',
+            'TINYINT',
+            'SMALLINT',
+            'INT',
+            'BIGINT'
+        ],
+        PostgreSQL: [],
+        Presto: []
+    };
+
+    export const REALS: DataTypeList = {
+        SQL: ['REAL'],
+        Oracle: [
+            'NUMBER(p?, s?)',
+            'BINARY_FLOAT',
+            'BINARY_DOUBLE'
+        ],
+        MySQL: [],
+        PostgreSQL: [],
+        Presto: [],
+        SQLServer: [
+            'REAL',
+            'FLOAT(n?)',
+            'DECIMAL(p?, s?)',
+            'NUMERIC(p?, s?)',
+            'SMALLMONEY',
+            'MONEY'
+        ]
+    };
+
+    export const BOOLEANS: DataTypeList = {
+        SQL: ['BOOLEAN', 'BOOL'],
+        MySQL: [],
+        Oracle: [],
+        PostgreSQL: [],
+        Presto: [],
+        SQLServer: []
+    };
+
+    export const BLOBS: DataTypeList = {
+        SQL: ['BLOB'],
+        MySQL: [
+            'BINARY(size)',
+            'VARBINARY(size)',
+            'TINYBLOB',
+            'BLOB(size)',
+            'MEDIUMBLOB',
+            'LONGBLOB'
+        ],
+        Oracle: [
+            'RAW(size)',
+            'LONG RAW',
+            'BLOB',
+            'BFILE'
+        ],
+        PostgreSQL: [],
+        Presto: [],
+        SQLServer: [
+            'BINARY(n)',
+            'VARBINARY',
+            'IMAGE'
+        ]
+    };
+
+    export const DATETIMES: DataTypeList = {
+        SQL: ['DATETIME'],
+        MySQL: [
+            'DATE',
+            'DATETIME(fsp?)',
+            'TIMESTAMP(fsp?)',
+            'TIME(fsp?)',
+            'YEAR'
+        ],
+        SQLServer: [
+            'DATETIME',
+            'DATETIME2',
+            'SMALLDATETIME',
+            'DATE',
+            'TIME',
+            'DATETIMEOFFSET',
+            'TIMESTAMP'
+        ],
+        Oracle: [
+            'TIMESTAMP(fsp?)',
+            'TIMESTAMP(fsp?) WITH TIME ZONE',
+            'TIMESTAMP(fsp?) WITH LOCAL TIME ZONE',
+            'INTERVAL YEAR(p?) TO MONTH',
+            'INTERVAL DAY(p?) TO SECOND'
+        ],
+        PostgreSQL: [],
+        Presto: []
+    };
+
+    const allStrings = all(STRINGS);
+    const allIntegers = all(INTEGERS);
+    const allReals = all(REALS);
+    const allDateTimes = all(DATETIMES);
+    const allBlobs = all(BLOBS);
+    const allBooleans = all(BOOLEANS);
+
+    function all(list: DataTypeList): string[] {
+        return distinct(Object.values(list).flat().map(getName));
```

Comment:
How to map back the types to a database again? We need to add validation rules and completion providers must also suppress 'wrong' types...